### PR TITLE
[cleanup] erefactor/AutoRefactor - Log parameters rather than log mes…

### PR DESCRIPTION
**Pull request by cal101**
…sage

Example PR for https://github.com/OpenAPITools/openapi-generator/issues/9097

AutoRefactor cleanup 'LogParametersRatherThanLogMessage' applied by erefactor:

Replaces a string concatenation as parameter of a logger method by a
string template followed by objects.

For AutoRefactor see https://github.com/JnRouvignac/AutoRefactor
For erefactor see https://github.com/cal101/erefactor

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ObjcClientCodegen.java [hunk 1]
```diff
@@ -435,7 +435,8 @@ public boolean isDataTypeBinary(String dataType) {
     public String toModelName(String type) {
         // model name cannot use reserved keyword
         if (reservedWords.contains(type)) {
-            LOGGER.warn(type + " (reserved word) cannot be used as model name. Renamed to " + ("model_" + type) + " before further processing");
+            LOGGER.warn("{} (reserved word) cannot be used as model name. Renamed to {} before further processing",
+					type, "model_" + type);
```
Comment (agilob) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ObjcClientCodegen.java:439]: this still can be improved to `{} {}` in log message

Comment (cal101) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ObjcClientCodegen.java:439]: Interesting case, may be worth improving the cleanup.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KtormSchemaCodegen.java [hunk 2]
```diff
@@ -688,7 +688,8 @@ public void processNullAndDefault(CodegenModel model, CodegenProperty property,
             try {
                 columnDefinition.put("colDefault", toColumnTypeDefault(defaultValue, dataType, dataFormat));
             } catch (RuntimeException exception) {
-                LOGGER.warn("Property '" + baseName + "' of model '" + model.getName() + "' mapped to data type which doesn't support default value");
+                LOGGER.warn("Property '{}' of model '{}' mapped to data type which doesn't support default value",
+						baseName, model.getName());
```
[the remark this answers is not in the document]
Comment (cal101) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KtormSchemaCodegen.java:692]: As I wrote in the related ticket, the cleanup breaks some formatting.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ObjcClientCodegen.java (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ObjcClientCodegen.java:439]: Good catch. We can improve it in another PR.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/MysqlSchemaCodegen.java [hunk 3]
```diff
@@ -364,7 +364,9 @@ public void processIntegerTypeProperty(CodegenModel model, CodegenProperty prope
             List<Object> enumValues = (List<Object>) allowableValues.get("values");
             for (int i = 0; i < enumValues.size(); i++) {
                 if (i > ENUM_MAX_ELEMENTS - 1) {
-                    LOGGER.warn("ENUM column can have maximum of " + ENUM_MAX_ELEMENTS.toString() + " distinct elements, following value will be skipped: " + (String) enumValues.get(i));
+                    LOGGER.warn(
+							"ENUM column can have maximum of {} distinct elements, following value will be skipped: {}",
+							ENUM_MAX_ELEMENTS.toString(), (String) enumValues.get(i));
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/MysqlSchemaCodegen.java:369]: @cal101 @agilob not sure why these spaces are added after the refactor. Will file a PR to clean it up manually later instead.

**Comment by wing328** [comment]
I can merge this as the refactoring uses tabs instead of 4-space (caught by Travis CI). Partial result below:
```
$ /bin/bash ./bin/utils/detect_tab_in_java_class.sh
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RubyClientCodegen.java:					camelize("model_" + modelName));
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java:					camelizedName);
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java:					camelize("call_" + operationId));
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PowerShellClientCodegen.java:					"{} with {} generator is ignored. Setting this value independently of {} is not currently supported.",
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PowerShellClientCodegen.java:					CodegenConstants.MODEL_PACKAGE, this.getName(), CodegenConstants.PACKAGE_NAME);
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PowerShellClientCodegen.java:					"{} with {} generator is ignored. Setting this value independently of {} is not currently supported.",
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PowerShellClientCodegen.java:					CodegenConstants.API_PACKAGE, this.getName(), CodegenConstants.PACKAGE_NAME);
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PowerShellClientCodegen.java:					name, camelize("model_" + name));
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PowerShellClientCodegen.java:					camelize("model_" + name));
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PowerShellClientCodegen.java:					escapeReservedWord(name));
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PowerShellClientCodegen.java:					escapeReservedWord(name));
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java:				"{} is hard coded to true in this generator. Alias models will only be generated if they contain validations or enums",
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java:				CodegenConstants.GENERATE_ALIAS_AS_MODEL);
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RClientCodegen.java:					camelize("model_" + name));
modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ProtobufSchemaCodegen.java:					camelize("model_" + name));
```

Can we change the refactoring tool to use 4-space instead of tabs?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/MysqlSchemaCodegen.java (hunk 3 above)
Comment (agilob) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/MysqlSchemaCodegen.java:369]: toString also should be removed from refactored code as LOGGER will call that internally

**Comment by cal101** [comment]
@wing328 Solving the wrong problem took much longer than expected. But solution may be useful for other cases. A bug in the cleanup was fixed and indentation is changed to 4 spaces.

**Review by wing328 — APPROVED** [review]
LGTM

**Pull request merged**